### PR TITLE
[dhcp_test] Extend test_dhcp_pkt_recv to support dualtor topo format

### PR DESCRIPTION
**Pull request by w1nda**
### Description of PR

Summary:
Fixes # (issue)
The format of host interfaces in dualtor topo file is quite different, we need to extend the code to support it.

For host interfaces in mo topo file is a number, however, in dualtor it's in format like 0.10@10,1.10@10 or 0.10,1.10

To support it, we use regex to match correct interface index

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [x] 202305
- [x] 202311
- [x] 202405

#### What is the motivation for this PR?
The format of host interfaces in dualtor topo file is quite different, we need to extend the code to support it.

#### How did you do it?
Use regex to match correct interface index

#### How did you verify/test it?
test it on mx, m0 dualtor and dualtor-aa testbed

### CODE REVIEW

**Review by yaqiangz — APPROVED** [review]

**Comment by w1nda** [comment]
Hi @wangxin, cloud you merge this PR? thanks

File: tests/dhcp_relay/test_dhcp_pkt_recv.py [hunk 1]
```diff
@@ -36,17 +37,27 @@ def check_dhcp_relay_feature_state(duthost):
...
     @pytest.fixture(scope="class")
-    def setup_teardown(self, duthost, tbinfo):
+    def setup_teardown(self, rand_selected_dut, tbinfo):
         disabled_host_interfaces = tbinfo['topo']['properties']['topology'].get('disabled_host_interfaces', [])
         ptf_indices = [interface for interface in tbinfo['topo']['properties']['topology'].get('host_interfaces', [])
                        if interface not in disabled_host_interfaces]
-        dut_intf_ptf_index = duthost.get_extended_minigraph_facts(tbinfo)['minigraph_ptf_indices']
+        dut_intf_ptf_index = rand_selected_dut.get_extended_minigraph_facts(tbinfo)['minigraph_ptf_indices']
+        if 'dualtor' in tbinfo['topo']['name']:
+            pattern = r'0.(\d+)'
```
Comment (vkjammala-arista) [inline tests/dhcp_relay/test_dhcp_pkt_recv.py:46]: I don't think this covers all the topologies. Ex: `0.0@0,1.0@0` (dualtor-120 topology)

Also hardcoding `0.` means we are considers first duthost (uppper tor), so theoretically this is incorrect (we need to get the ptf_indices for the rand_selected_dut).

File: tests/dhcp_relay/test_dhcp_pkt_recv.py [hunk 2]
```diff
@@ -36,17 +37,27 @@ def check_dhcp_relay_feature_state(duthost):
...
+            ptf_indices = [int(re.findall(pattern, index)[0]) for index in ptf_indices if re.match(pattern, index)]
         yield ptf_indices, dut_intf_ptf_index
 
-    def test_dhcpv6_multicast_recv(self, duthost, ptfadapter, setup_teardown):
+    def test_dhcpv6_multicast_recv(
+        self,
+        rand_selected_dut,
+        setup_standby_ports_on_rand_unselected_tor,
```
Comment (vkjammala-arista) [inline tests/dhcp_relay/test_dhcp_pkt_recv.py:53]: In addition to this, We need `toggle_all_simulator_ports_to_rand_selected_tor` fixture (needed for `active-standby` topologies)

Comment (vkjammala-arista) [reply tests/dhcp_relay/test_dhcp_pkt_recv.py:53]: Also other fixtures in this module (Ex: setup_teardown_acl) need to have `rand_selected_dut` fixture.

**Pull request closed without merging**